### PR TITLE
Wiki to doc edit4

### PR DESCRIPTION
**Pull request by ralfjansson**
text about work arounds migrating custom effects from XNA to MonoGame.

### CODE REVIEW

**Comment by ralfjansson** [comment]
Hope this PR worked byyert than my last one. Tried to get the latest version Before making the PR.

**Comment by ralfjansson** [comment]
Also edited the original wiki page with a reference to the new documentation.

**Comment by ralfjansson** [comment]
When I tried commiting this change the GitHub desktoptool complained about a ThirdParty/Dependencies file having been changed. I avoided commitning this change to be able to commit my other changes. Can someone explain why the tool or git makes this change when creating a branch to work on? It just appears from 'nowhere'...
Looks like this:
@@ -1 +1 @@
- Subproject commit 0405edfb770200d30b0921a2484f01ddcb70a230
+ Subproject commit 69585dd967ae453f4af0d43177d397cdf7bf4bc3

**Comment by polsys** [comment]
That's because the Dependencies folder is implemented as a submodule, which are not automatically updated when fetching the branch. Run `git submodule update` on the Git shell, so that you get the latest version.

**Comment by ralfjansson** [comment]
ok. Thank you!

File: Documentation/custom_effects.md [hunk 1]
```diff
@@ -48,6 +48,104 @@ These are some tips for writing or converting effects for use with MonoGame.
...
+Look at the code on line 29 of MGEffectProcessor.cs! 
+
+`options.DX11Profile = platform == MonoGamePlatform.Windows8 ? true : false; `
+
+
+
+The **ContentHelper.cs** file in the ContentProcessors project reads 
+the build configuration name you are using then does this:
+
+`switch (platform.ToUpper())`
```
Comment (tomspilman) [inline Documentation/custom_effects.md]: When writing a block of code wrap it in the triple single quotes...  do not wrap each line individually.

File: Documentation/custom_effects.md [hunk 2]
```diff
@@ -48,6 +48,104 @@ These are some tips for writing or converting effects for use with MonoGame.
 * The effect compiler is aggressive about removing unused paramters, be sure the parameters you are setting are actually used.
 * If you think you've found a bug porting a shader [please let us know](https://github.com/mono/MonoGame/issues).
 
+### Summary of some work arounds in the process of transferring effects files across to MonoGame from XNA
```
Comment (tomspilman) [inline Documentation/custom_effects.md]: Try to keep header titles short and simple.

File: Documentation/custom_effects.md [hunk 3]
```diff
@@ -48,6 +48,104 @@ These are some tips for writing or converting effects for use with MonoGame.
...
+
+Otherwise you may get **"Wrong MGFX file version!"** raised by ReadEffect() in Effect.cs when you do a **content.Load<Effect>**
+[because the version written in to the .xnb file is not a match with the value of MGFXVersion 
+that has been put in the run time].
+
+So if you have this error, the first thing to check is that your content processor is the same up-to-dateness as the version of the runtime you are running your game against.
+
+
+#### Shader model compilation:
+(a) For **OpenGL** versions of MonoGame (eg. Mac, Linus, WindowsGL) the shader model must be **SM 3.0 or lower**
```
Comment (tomspilman) [inline Documentation/custom_effects.md]: Please stop using `(a)`, `(b)`, `(c)` to itemize things.  There are good ways to do this [via markdown (read this)](https://guides.github.com/features/mastering-markdown/).

File: Documentation/custom_effects.md [hunk 4]
```diff
@@ -48,6 +48,104 @@ These are some tips for writing or converting effects for use with MonoGame.
...
+run against the Windows DirectX version of MonoGame, you **MUST** use build profile that
+is **specifically called "Windows8" !!**  ... 
+either the one supplied with the project code or one you make yourself.
+
+At some point this getting written into a registry key  
+called "MONOGAME_PLATFORM" (HKEY_CURRENT_USER/Environment/MONOGAME_PLATFORM),
+and then gets read back from that key via:
+`var platform = Environment.GetEnvironmentVariable("MONOGAME_PLATFORM", EnvironmentVariableTarget.User);`
+
+If you don't realize this, having just converted all your .fx files to run against
```
Comment (tomspilman) [inline Documentation/custom_effects.md]: Don't put newlines in the middle of sentences...    markdown is not like HTML... it cares about newlines.  You end up with text that wraps prematurely which looks poor and wastes space.

**Comment by tomspilman** [comment]
So my main issue with this PR is that it is all formatted very poorly.

We have to provide documentation that is informative and easy to read.  For example:

 https://msdn.microsoft.com/en-us/library/windows/apps/mt210779.aspx

It has section titles that are simple and informative.  It has paragraphs of text.  It has lists where it needs them using proper bullet formatting.  It uses code block to display code legibly.  It has punctuation in the right places.

If you look at the doc before your changes:



Then look at the doc after:

  https://github.com/ralfjansson/MonoGame/blob/wiki-to-doc-edit4/Documentation/custom_effects.md

You can clearly see the problems.

**Comment by ralfjansson** [comment]
I agree. Shall we scrap this PR and let me redo the the job in a new PR?

**Comment by tomspilman** [comment]
You can just fix it here... it wouldn't be a problem.

**Comment by ralfjansson** [comment]
Ok.

**Comment by ralfjansson** [comment]
Reformatted my last changes and commited them. Tried to follow @tomspilman advice.

File: Documentation/custom_effects.md [hunk 5]
```diff
@@ -45,9 +45,95 @@ These are some tips for writing or converting effects for use with MonoGame.
 * On DirectX platforms use the `SV_Position` semantic instead of `POSITION` in vertex shader inputs.
 * Note that on GL platforms default values on Effect parameters do not work.  Either set the parameter from code or use a real constant like a #define.
 * Do not name your sampler `Sampler` - it will not compile.
-* The effect compiler is aggressive about removing unused paramters, be sure the parameters you are setting are actually used.
+* The effect compiler is aggressive about removing unused parameters, be sure the parameters you are setting are actually used.
 * If you think you've found a bug porting a shader [please let us know](https://github.com/mono/MonoGame/issues).
 
+### Transferring effects files from XNA to MonoGame
```
Comment (tomspilman) [inline Documentation/custom_effects.md]: Change this to ##.

File: Documentation/custom_effects.md [hunk 6]
```diff
@@ -45,9 +45,95 @@ These are some tips for writing or converting effects for use with MonoGame.
...
+
+* Aligning version of the content processor and the runtime.
```
Comment (tomspilman) [inline Documentation/custom_effects.md]: Change this to a header...  ###.

File: Documentation/custom_effects.md [hunk 7]
```diff
@@ -45,9 +45,95 @@ These are some tips for writing or converting effects for use with MonoGame.
...
+
+* Shader model compilation.
+	* For OpenGL versions of MonoGame (eg. Mac, Linus, WindowsGL) the shader model must be SM 3.0 or lower
+	* For DirectX versions of MonoGame (eg. WinRT, Windows8) the shader model must be SM 4.0 or higher.
+
+	In the DirectX case it is typical to use the 'vs_4_0_level_9_1'  and 'ps_4_0_level_9_1' compilation profiles, because these provide backwards compatibility for Shader Model 2 shaders.
+
+
+
+* Windows8 configuration.
```
Comment (tomspilman) [inline Documentation/custom_effects.md]: Change this to a header...  ###.

File: Documentation/custom_effects.md [hunk 8]
```diff
@@ -45,9 +45,95 @@ These are some tips for writing or converting effects for use with MonoGame.
...
+"Vertex shader 'SimpleVS' must be SM 3.0 or lower!"  
+
+
+ The "Windows" build configuration is used for the WindowsGL target.
+
+ Aside: The WindowsGL version of MonoGame is primarily intended to be a test environment. 
+Don't necessarily expect great performance from this build.
+
+
+* Windows8, DX11 profile, switch POSITION0 to SV_POSITION (in .fx file)
```
Comment (tomspilman) [inline Documentation/custom_effects.md]: Remove this section...  is it no longer relevant.

**Comment by tomspilman** [comment]
So unless we have someone that can rewrite this information, fixing the mistakes, removing dumb stuff (why is it giving a tour of the content pipeline implementation instead of telling the user how to port shaders), formatting it correctly and making it generally readable...  then this information shouldn't be merged.

I tried to help try to get things going here, but it is pretty clear to me that you need to know the subject matter to do this work.  If you don't it never turns out well.  My fault for not realizing this earlier.

**Pull request closed without merging**